### PR TITLE
Prevent expanding an object when a string is selected

### LINKED ISSUE #885: Do not toggle object inspector nodes when selecting text
1. Open the console
2. Enter `inspect({a: () => {}})`
```
▼ {…}
  ▶ a: function a()
​  ▶ __proto__: Object { … }
```
3. Select `fun`

Before you can e.g. copy it, the property is expanded and `fun` disappears:
```
▼ {…}
  ▼ a: a()
     length: 0
     name: "a"
​  ▶ __proto__: Object { … }
```

### PR DESCRIPTION
**Pull request by nchevobbe**
_Previously at https://github.com/devtools-html/devtools-core/pull/898 , finishing it since Sean might not have the time to do so_

---

Fix #885 

Prevent expanding an object when text is selected.
Add tests to make sure we don't expand a node, or trigger `onLabelClick`, when the
user selected some text.
Add a mock for `window.getSelection` since it is not defined in nodejs.

### Test Plan

- In `devtools-core/packages/devtools-reps` do `yarn install && yarn start` 
- open localhost:8000.
- Launch a new Firefox window
- select the tab in the launchpad
- evaluate `location`
- select the location. **It is important your mouseup is fired on the text itself**
- assert that the node wasn't expanded
- click on the node
- assert that the node was expanded

### CODE REVIEW

**Comment by nchevobbe** [comment]
@Loirooriol  do you mind having a look since you commented on the previous PR ?
I used Selection.type since it's the most supported property from what I saw on MDN

**Comment by Loirooriol** [comment]
Comparing http://help.dottoro.com/ljitmswc.php and http://help.dottoro.com/ljlflvjl.php I'm not sure `Selection.type` is more supported, but at least it appears in the spec.

I get various errors like `Module not found: Error: Can't resolve 'react/lib/getNextDebugID'` when I use `yarn install && yarn start` so I can't test, but the patch seems OK to me.

File: packages/devtools-reps/src/object-inspector/index.js [hunk 1]
```diff
@@ -407,6 +412,13 @@ class ObjectInspector extends Component {
             onClick: onLabelClick
               ? event => {
                 event.stopPropagation();
+
+                const selection = getSelection();
+                // If the user selected text, bail out.
+                if (selection && selection.type === "Range") {
+                  return;
+                }
+
```
Comment (janodvarko) [inline packages/devtools-reps/src/object-inspector/index.js]: nit: you might introduce a new `this.getSelection` method that wraps the logic.

File: packages/devtools-reps/src/object-inspector/tests/component/events.js [hunk 2]
```diff
@@ -80,4 +80,32 @@ describe("ObjectInspector - properties", () => {
 
     expect(onLabelClick.mock.calls.length).toBe(1);
   });
+
+  it("does calls the onLabel prop function when the user selected text", () => {
```
Comment (janodvarko) [inline packages/devtools-reps/src/object-inspector/tests/component/events.js]: typo: does call the ...

Comment (nchevobbe) [reply packages/devtools-reps/src/object-inspector/tests/component/events.js]: fixed

File: packages/devtools-reps/src/object-inspector/index.js (hunk 1 above)
Comment (nchevobbe) [reply packages/devtools-reps/src/object-inspector/index.js]: good idea, I created a util for that (`documentHasSelection`)

**Pull request merged**